ensure that $BUSSER_ROOT is owned by the current user...

tknerr opened the pull request:
...to compensate for permission issues with vagrant synched folders

This PR is meant to work around the [permission issue mentioned here](https://github.com/test-kitchen/test-kitchen/issues/671#issuecomment-87449673) which occurs if parts of `$BUSSER_ROOT` (e.g. `/tmp/verifier/gems/cache`) are mounted via vagrant-cachier / vagrant synched folders.

I really don't like this workaround and would be happy if you decline it. Really. I mainly created the PR to get feedback on better ways / places to fix it.

**Considerations:**
- Since vagrant-cachier does not support windows guests I have only considered the `busser_install_command.sh`, not the .bat variant
- Initially wanted to place that fix somewhere in kitchen-vagrant (since it fixes a problem that occurs in a vagrant specific setup), but I decided against it because kitchen-vagrant does not know anything about `$BUSSER_ROOT`
- :warning: this might break things if the `$USER`'s group is not `$USER`!

fnichol commented:
@tknerr Thanks for your feedback on this. I'll see what we can do this week to make the setup more cachier-friendly. The issue that we're dealing with is that now the SSH Transport SCPs the files over as the non-root user and thus wouldn't have permission to create the `$BUSSER_ROOT/suites`. Does cachier have to have root own all directories going back up to root on the synced folders?

tknerr commented:
@fnichol it's actually how vagrant synced folders work, not vagrant-cachier itself:

> The permissions you specify on shared folders will only apply to the deepest child. Otherwise, typical umask takes place

(from here: https://github.com/mitchellh/vagrant/issues/2257#issuecomment-24906376)

Since I wanted to make vagrant-cachier both optional and transparent (i.e. without changing .kitchen.yml), I added caching via my global `~/.vagrant.d/Vagrantfile`. I also added a shell provisioner there to fix the permissions, but it does not run since kitchen-vagrant creates the VM with the `--no-provision` flag. 

Hmmm... I just saw there is a `provision: false` config option in kitchen-vagrant. 
Oh. And I just saw there is a global `~/.kitchen/config.yml` as well!

I could probably combine the above to make it work just with the global vagrant & kitchen config. That way it would be completely external and we wouldn't need to change anything in test-kitchen or the kitchen-vagrant driver. Let me try...

tknerr commented:
Looks good! (I should have rubber-ducked this scenario before, sorry ;-))

So now I have in my `~/.vagrant.d/Vagrantfile`:
```ruby
Vagrant.configure("2") do |config|

  # enable cachier globally
  if Vagrant.has_plugin?("vagrant-cachier")
    config.cache.scope = :box

    # cache bussers for test-kitchen
    config.cache.enable :generic, {
      "verifier-gems" => { cache_dir: "/tmp/verifier/gems/cache" }
    }
    # correct permissions
    config.vm.provision "shell", inline: "chown -R vagrant:vagrant /tmp/verifier"
  end
end
```

And in `~/.kitchen/config.yml`:
```yaml
---
driver:
  name: vagrant
  # run provisioners (e.g. from global Vagrantfile)
  provision: true
```

However, this creates an easily overlooked soft dependency between the two files. Ideally I'd drive that specific kitchen-vagrant configuration from within the global Vagrantfile.

I have seen no env var like `KITCHEN_VAGRANT_PROVISION` or such that I could set from within the Vagrantfile, is there? (But it probably wouldn't help anyway as it's likely too late then already...)

Or any other ideas how to drive the kitchen-vagrant config from within the Vagrantfile?

tknerr commented:
Also: is it correct to cache only `/tmp/verifier/gems/cache`?

I was about to cache the whole `/tmp/verifier/gems` directory, but that would probably opt me out from any busser updates (as [per this](https://github.com/test-kitchen/test-kitchen/blob/4f2e2189937cb2e83ef173e3b48aef46a05ef1fa/support/busser_install_command.sh#L5-7)) once it's in the cache, I suspect.
